Update README.md (#43)
ThePokeu/EnChord: README.md

# EnChord

> **EnChord** is a perfect tool to develop your music dictation skills. Challenge yourself with different levels of quizzes and follow your progress by checking your highest score and average test results. You can also review what are the frequently mistaken interval types.

Reviewer: I'm probably misunderstanding it, but I think dictation refers to writing down what you hear - would "aural skills" be a better fit here?

Author: That is actually true! Musical dictation mostly includes marking down what you hear. I will update it now. Also, I was thinking to include some of the issues you raised in #40 in this PR,  except changing the deployed website name. What do you think? I thought, these are too minors (doesn't mean they are not important) to make another PR 😄

Author: (You gotta sell what you made regardless of the quality right 🤣 🤣 . So it is a perfect tool for me. Or at least will be. haha)

Reviewer: Thanks for updating the project name and checking off the relevant boxes ❤️

## Table of contents

> 1. [How it works](https://github.com/TheChopsticks/EnChord/edit/main/README.md#how-it-works)

Reviewer: ```suggestion
1. [How it works](#how-it-works)
```
Since we're linking within the same document, just using the anchor should be sufficient

Author: Thank you! I will change them now. Apparently, I love taking a rough road 🤣

2. [Future feature](https://github.com/TheChopsticks/EnChord/edit/main/README.md#future-features)
3. [How EnChord was built](https://github.com/TheChopsticks/EnChord/edit/main/README.md#how-enchord-was-built)
4. [Installation](https://github.com/TheChopsticks/EnChord/edit/main/README.md#installation)
5. [Credits](https://github.com/TheChopsticks/EnChord/edit/main/README.md#credits)
6. [License](https://github.com/TheChopsticks/EnChord/edit/main/README.md#license)
7. [Badges](https://github.com/TheChopsticks/EnChord/edit/main/README.md#badges)

## How it works

> #### Try 3 different levels

Reviewer: I was wondering why we jumped from a level 2 (`##`) to a level 4 (`####`) heading here. Could we use a level 3 instead, or is it not a suitable choice here?

Author: I didn't really think it through here. I changed them now!

Author: I didn't really think it through here. I changed them now!

- **EnChord** has 3 different levels, Easy, Intermediate, and Hard.

### What are the differences

- **For enthusiastic beginners:**

  - 2 random notes will be played only upwards to help you guess the answer.
  - There is an unlimited number of getting hints for you to listen to full scale between 2 notes

- **For intermediate-level players with basic skills:**

  - 2 random notes will be played either upwards or downwards
  - There is a limited number of hints

- **For skilled musicians:**
  - 2 random notes will be played TOGETHER
  - There is no hint available.

## Future features

### Review and follow your progress:

- After each quiz set, you can review which questions you got wrong and listen to them again!
- Check your highest score and the average score of previously played games.
- Check your weakness by reviewing the most wrong interval types.

> #### Language support:
> 

Reviewer: ```suggestion
### Language support:

```

> - Mandarin(普通话), Korean(한국어), Hungarian(Magyar)

Reviewer: ```suggestion
- Hungarian (Magyar), Korean (한국어), Traditional Chinese (繁體中文)
```
Nitpick on the terminology here ("Mandarin" is usually used to refer to the spoken language) and a small suggestion to use alphabetic order so we don't show unintended preference 🙃

Author: Actually, I wanted to ask you what to write for the Chinese part. I googled them but still wasn't 100% sure. Thank you. Also now I know, it is a norm to use alphabetic order 👍

## How EnChord was built

> The `EnChord` is the very first project of `The Forks`. This app is written in plain JS and CSS to improve our `VanilaJS` and `VanilaCSS` skills and to take a deep dive into code organization. To write clean and organized code without any help of frameworks, we introduced some design patterns.

Reviewer: ```suggestion
The `EnChord` is the very first project of `The Pokeu`. This app is written in plain JS and CSS to improve our `Vanilla JS` and `Vanilla CSS` skills and to take a deep dive into code organization. To write clean and organized code without any help of frameworks, we introduced some design patterns.
```

Author: ❤️

- First was the [MVC](https://developer.mozilla.org/en-US/docs/Glossary/MVC) architecture. We separated the pure game logic(Models) and the UI(Views), and created the `controller` class to manage the communication between other modules.

> - To achieve this "separation of concerns", we implemented another design pattern, [Publish/Subscribe](https://ably.com/topic/pub-sub). Sharing common knowledge through a data pipe, the `controller` class, other modules can sorely focus on performing their tasks and do not need to worry about whereabouts of inputs and outputs.

Reviewer: ```suggestion
- To achieve this "separation of concerns", we implemented another design pattern, [Publish/Subscribe](https://ably.com/topic/pub-sub). Sharing common knowledge through a data pipe, the `controller` class, other modules can solely focus on performing their tasks and do not need to worry about whereabouts of inputs and outputs.
```

> - First was the [MVC](https://developer.mozilla.org/en-US/docs/Glossary/MVC#:~:text=MVC%20(Model%2DView%2DController,of%20labor%20and%20improved%20maintenance.) architecture. We separated the pure game logic(Models) and the UI(Views), and created the `controller` class to manage the communication between other modules.

Reviewer: ```suggestion
- First was the [MVC](https://developer.mozilla.org/en-US/docs/Glossary/MVC) architecture. We separated the pure game logic(Models) and the UI(Views), and created the `controller` class to manage the communication between other modules.
```
I think this shorter version of the link works too 😄

Author: 👍  Thank you

The benefits of such patterns helped us to avoid spaghetti code and made it easy to maintain the app and expand game features.

> As our aim was to develop an app that is more than just a one-time use, we also felt the need of having a database system. Therefore, we utilized the [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) to store scores of users and have stored data available for [future features](https://github.com/TheChopsticks/EnChord/edit/main/README.md#future-features) . Considering future scalability of the application, `storage` module is written in a way that the change of storage method is possible.

Reviewer: ```suggestion
As our aim was to develop an app that is more than just a one-time use, we also felt the need of having a database system. Therefore, we utilized the [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) to store scores of users and have stored data available for [future features](#future-features) . Considering future scalability of the application, `storage` module is written in a way that the change of storage method is possible.
```
Relative link is also fine here

To build the project, run:
`npm run build`

## Credits

- [Utopia](https://utopia.fyi/)

## License

_EnChord_ is available under the MIT license. See the LICENSE file for more information.

## Badges

> [![Netlify Status](https://api.netlify.com/api/v1/badges/61aa5294-7ef4-4d3d-978c-b61c9ac5fe4e/deploy-status)](https://app.netlify.com/sites/enchord/deploys)

Reviewer: Instead of putting badges in their own section, we could also think about putting them closer to the top of the README, so people can easily get an overview of statuses for this project.

An example:
<img width="613" alt="Screenshot 2022-12-09 at 05 26 21" src="https://user-images.githubusercontent.com/62935115/206623925-c477706e-dc60-4a09-b991-ec40473c959b.png">

Author: That is indeed a better idea!
